clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy

```diff
@@ -174,16 +173,22 @@ class DeployKubernetesAtomicOperation implements AtomicOperation<DeploymentResul
     return replicaSet
   }
 
-  HasMetadata deployController(KubernetesV1Credentials credentials, String controllerName, String clusterName, String namespace) {
+  HasMetadata deployController(KubernetesV1Credentials credentials, def serverGroupNameResolver, String clusterName, String namespace) {
     def controllerSet
-    def canUpdated = KubernetesClientApiConverter.canUpdated(description)
+    def isUpdateControllerEnabled = KubernetesClientApiConverter.isUpdateControllerEnabled(description)
+    def controllerName
+    if (KubernetesClientApiConverter.validateSequence(description)) {
```

Comment:
If validation fails, shouldn't that be a user error? e.g. the sequence desired can't be deployed?

Reply:
Then, it should not use sequence which it will go to else.